packages/compass-sidebar/src/components/navigation-items.tsx

```diff
@@ -245,31 +252,45 @@ export function NavigationItems({
         showTooManyCollectionsInsight={showTooManyCollectionsInsight}
       />
       {isExpanded && (
-        <DatabaseCollectionFilter changeFilterRegex={changeFilterRegex} />
+        <DatabaseCollectionFilter onFilterChange={onFilterChange} />
       )}
       {isExpanded && <SidebarDatabasesNavigation />}
     </>
   );
 }
 
-const mapStateToProps = (state: RootState) => {
+const mapStateToProps = (
+  state: RootState,
+  { readOnly: preferencesReadOnly }: { readOnly: boolean }
+) => {
   const totalCollectionsCount = state.databases.databases.reduce(
     (acc: number, db: { collectionsLength: number }) => {
       return acc + db.collectionsLength;
     },
     0
   );
+  const instanceFetched = ['refreshing', 'ready'].includes(
+    state.instance?.status ?? ''
+  );
+  const isDataLake = instanceFetched
+    ? state.instance?.dataLake.isDataLake
+    : true;
```

Comment:
This is a total nit, but since you're making the assumption that something is datalake when the instance isn't fetched so that the perf item won't show by default it is probably better/safer to just name this temp var showPerformanceItem immediately. Just in case someone comes along in future and uses isDataLake to display something specific to datalake and in that case you probably want false as a safer default?

Although that would complicate showCreateDatabaseAction below where you use isDataLake again, so maybe more trouble than its worth 🤔

Reply:
Hmmm, yeah, good point, I feel like I can just move this logic to a function where default value is provided and then it's not that complicated to use and the intent is explicit

Reply:
Refactored this a bit in https://github.com/mongodb-js/compass/pull/5169/commits/60f022de60087e4d53faf2407d61c07ca10f6702. A bit more vebose, but probably not a bad thing?